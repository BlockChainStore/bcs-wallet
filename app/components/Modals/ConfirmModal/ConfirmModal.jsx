// @flow
import React from 'react'

import BaseModal from '../BaseModal'
import Button from '../../Button'
import styles from './ConfirmModal.scss'

type Props = {
    title: string,
    text: string,
    onClick: Function,
    onCancel: Function,
    hideModal: Function,
    width: string,
    height: string,
}

const ConfirmModal = ({ hideModal, title, onClick, onCancel, text, width, height }: Props) => (
  <BaseModal
    title={title}
    hideModal={hideModal}
    style={{
      content: {
        width,
        height
      }
    }}
  >
    <div className={styles.textContainer}>
      <strong className={styles.text}>{text}</strong>
    </div>
    <div className={styles.modalFooter}>
      <Button
        id='confirm'
        className={styles.actionButton}
        onClick={() => {
          onClick()
          hideModal()
        }}>Confirm</Button>
      <Button
        id='cancel'
        cancel
        onClick={() => {
          hideModal()
          if (onCancel) {
            onCancel()
          }
        }}>Cancel</Button>
    </div>
  </BaseModal>
)

ConfirmModal.defaultProps = {
  width: '450px',
  height: '175px'
}

export default ConfirmModal
