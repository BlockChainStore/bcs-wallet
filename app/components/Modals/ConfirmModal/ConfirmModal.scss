.modalFooter {
    padding-top: 20px;
    text-align: right;
}

.actionButton {
    background-color: red;
    &:hover {
        background-color: red;
    }
}

.cancelButton {
    background: none;
    border: 1px solid #999;
    color: #000;
    &:hover {
        background: none
    }
}
