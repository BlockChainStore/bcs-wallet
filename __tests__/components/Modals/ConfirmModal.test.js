import React from 'react'
import { shallow } from 'enzyme'
import ConfirmModal from '../../../app/components/Modals/ConfirmModal/ConfirmModal'

describe('ConfirmModal', () => {
  const props = {
    title: 'test modal',
    text: 'text',
    hideModal: jest.fn(),
    onClick: jest.fn(),
    onCancel: jest.fn()
  }

  test('should render without crashing', () => {
    const wrapper = shallow(<ConfirmModal {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('should render the title correctly', () => {
    const wrapper = shallow(<ConfirmModal {...props} />)

    expect(wrapper.dive().find('.modalHeaderTitle').text()).toEqual(props.title)
  })

  test('should render the text correctly', () => {
    const wrapper = shallow(<ConfirmModal {...props} />)

    expect(wrapper.dive().find('.text').text()).toEqual(props.text)
  })

  test('should trigger the onCancel function followed by hideModal', () => {
    const wrapper = shallow(<ConfirmModal {...props} />)

    wrapper.dive().find('#cancel').simulate('click')

    expect(props.onCancel).toHaveBeenCalled()
    expect(props.hideModal).toHaveBeenCalled()
  })

  test('should trigger the onClick function followed by hideModal', () => {
    const wrapper = shallow(<ConfirmModal {...props} />)

    wrapper.dive().find('#confirm').simulate('click')

    expect(props.onClick).toHaveBeenCalled()
    expect(props.hideModal).toHaveBeenCalled()
  })
})
