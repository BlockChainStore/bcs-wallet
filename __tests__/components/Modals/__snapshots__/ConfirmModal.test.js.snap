// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal should render without crashing 1`] = `
<BaseModal
  height="450px"
  hideModal={[Function]}
  style={
    Object {
      "content": Object {
        "height": "175px",
        "width": "450px",
      },
    }
  }
  title="test modal"
  width="450px"
>
  <div
    className="textContainer"
  >
    <strong
      className="text"
    >
      text
    </strong>
  </div>
  <div
    className="modalFooter"
  >
    <Button
      cancel={false}
      className="actionButton"
      id="confirm"
      onClick={[Function]}
      primary={true}
      secondary={false}
    >
      Confirm
    </Button>
    <Button
      cancel={true}
      id="cancel"
      onClick={[Function]}
      primary={true}
      secondary={false}
    >
      Cancel
    </Button>
  </div>
</BaseModal>
`;
